HouseRules_Essentials/Rules/BackstabConfigOverriddenRule.cs

```diff
@@ -0,0 +1,59 @@
+﻿namespace HouseRules.Essentials.Rules
+{
+    using System.Collections.Generic;
+    using Boardgame;
+    using Boardgame.BoardEntities;
+    using DataKeys;
+    using HarmonyLib;
+    using HouseRules.Types;
+    using StatusEffectData = global::Types.StatusEffectData;
+
+    public sealed class BackstabConfigOverriddenRule : Rule, IConfigWritable<List<BoardPieceId>>, IPatchable, IMultiplayerSafe
+    {
+        public override string Description => "Backstab is allocated to new BoardPieceIDs";
+
+        private static bool _isActivated;
+        private static List<BoardPieceId> _adjustments;
+
+        public BackstabConfigOverriddenRule(List<BoardPieceId> adjustments)
+        {
+            _adjustments = adjustments;
+        }
+
+        public List<BoardPieceId> GetConfigObject() => _adjustments;
+
+        protected override void OnActivate(GameContext gameContext)
+        {
+            _isActivated = true;
+        }
+
+        protected override void OnDeactivate(GameContext gameContext) => _isActivated = false;
+
+        private static void Patch(Harmony harmony)
+        {
+            harmony.Patch(
+                original: AccessTools.Method(typeof(Piece), "IsRogue"),
+                prefix: new HarmonyMethod(
+                    typeof(BackstabConfigOverriddenRule),
+                    nameof(Piece_IsRogue_Prefix)));
+        }
+
+        private static bool Piece_IsRogue_Prefix(ref Piece __instance, ref bool __result)
+        {
+            if (!_isActivated)
+            {
+                return true;
+            }
+
+            if (_adjustments.Contains(__instance.boardPieceId))
```

Comment:
> I'm guessing that there is probably a better way of doign this so that rules don't overwrite each other's settings.... Possibly with the patch to make isRogue take a list of IDs being separate from the rule that configures the list... but that's beyond my ability to implement at the moment.

Does the following info help?

---

We may want to introduce a static/global field (e.g., like `_globalMultiplier` in `CardEnergyFromRecyclingMultipliedRule`), instead of using `_adjustments`, so that multiple versions of this rule don't interfere with each other.  I think all other `IPatchable` rules already use some `_global` field.

Example issue:
- Ruleset created and uses `new BackstabConfigOverriddenRule(myFieldsOne);`.  That sets the static `_adjustments` to `myFieldsOne`.
- Another ruleset created and uses `new BackstabConfigOverriddenRule(myFieldsTwo);`.  That sets (**_overrides_**) the static `_adjustments` to `myFieldsTwo`
- Even if the first ruleset is loaded, the latest instantiation of the rule (using `myFieldsTwo`) will apply.

Using a static `_global` field is an unfortunate (but simple/elegant) workaround and originates from a limitation in Harmony that a patch method must be static.  Static patch method implies using static fields.

Reply:
Thank you - that was exactly what I was looking for. I've now added a similar _globalAdjustments and given it a test. All appears to work correctly.

Reply:
I decided that the rulename was OK for this. It is not adjusting a PieceConfig property, and doesn't break our naming convention for ability rules. I'm gonna merge now.